src/TSTransformer/nodes/statements/transformReturnStatement.ts

```diff
@@ -14,30 +15,58 @@ function isTupleReturningCall(state: TransformState, tsExpression: ts.Expression
 	);
 }
 
-export function transformReturnStatementInner(state: TransformState, returnExp: ts.Expression) {
-	let expression: luau.Expression | luau.List<luau.Expression> = transformExpression(state, skipDownwards(returnExp));
-	if (isLuaTupleType(state)(state.getType(returnExp)) && !isTupleReturningCall(state, returnExp, expression)) {
-		if (luau.isArray(expression)) {
-			expression = expression.members;
-		} else {
-			expression = luau.call(luau.globals.unpack, [expression]);
+function isTupleMacro(state: TransformState, expression: ts.Expression) {
+	if (ts.isCallExpression(expression)) {
+		const expType = state.typeChecker.getNonOptionalType(state.getType(expression.expression));
+		const symbol = getFirstDefinedSymbol(state, expType);
+		if (symbol && symbol === state.services.macroManager.getSymbolOrThrow(SYMBOL_NAMES.$tuple)) {
+			return true;
+		}
```

Comment:
Careful of code duplication, could this be wrapped by a MacroManager function?

Reply:
That quickly gets hairy. Might need to be a separate PR.